chain/registry.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chain
+
+import "github.com/ava-labs/hypersdk/codec"
+
+type Registry interface {
+	ActionRegistry() *codec.TypeParser[Action]
+	AuthRegistry() *codec.TypeParser[Auth]
+	OutputRegistry() *codec.TypeParser[codec.Typed]
+}
```

Comment:
Why do we need an interface for this? I think we can avoid this and just export `registry` + the fields on it.